crt_portal/cts_forms/views.py

```diff
@@ -387,25 +387,41 @@ def post(self, request):
 
 
 class ResponseView(LoginRequiredMixin, View):
+    """
+    Allow intake specialists to print, copy, or email form response letters
+    If we encounter _any_ exceptions in sending an email, log the error message and return.
+    """
+    MAIL_SERVICE = "AWS Simple Email Service"
+    ACTIONS = {'send': 'Emailed',
+               'copy': 'Copied',
+               'print': 'Printed'}
+    SEND_MAIL_ERROR = "There was a problem sending the requested email. No email was sent. We've logged this error and will review it as soon as possible."
 
     def post(self, request, id):
         report = get_object_or_404(Report, pk=id)
         form = ResponseActions(request.POST, instance=report)
+        url = preserve_filter_parameters(report, request.POST)
 
         if form.is_valid() and form.has_changed():
-            template_name = form.cleaned_data['templates'].title
+
+            template = form.cleaned_data['templates']
             button_type = request.POST['type']
-            actions = {
-                'send': 'Emailed',
-                'copy': 'Copied',
-                'print': 'Printed'
-            }
-            action = actions[button_type]
-            description = f"{action} '{template_name}' template"
-            add_activity(request.user, "Contacted complainant:", description, report)
+
+            if button_type == 'send':  # We're going to send an email!
+                try:
+                    crt_send_mail(report, template)
+                    description = f"Email sent: '{template.title}' to {report.contact_email} via {self.MAIL_SERVICE}"
+                except Exception as e:  # catch *all* exceptions
+                    logger.warning({'message': f"Email failed to send: {e}", 'report': report.id})
```

Comment:
I would love to emit these to NR, using something like https://docs.newrelic.com/docs/agents/python-agent/python-agent-api/recordexception-python-agent-api,  for easier alerting but not sure how to test locally

Reply:
https://docs.newrelic.com/docs/agents/python-agent/configuration/python-agent-configuration#developer_mode might be helpful here by allowing us to initialize and interact with a dummy agent locally